Leave neighborhood route

nina-dk opened the pull request:
I created the route for leaving a neighborhood and corresponding tests. We'll still need to do some sifting on the front-end to required confirmation from the admin for leaving as we discussed.

radu-constantin commented on apps/backend/src/services/neighborhoodServices.ts:
```diff
@@ -219,6 +257,76 @@ const connectUserToNeighborhood = async (userId: number, neighborhoodId: number)
+    error.name = 'Unauthorized';
+    throw error;
+  } else if (userIsNeighborhoodAdmin) {
+    // In this case, prompt the user that this action will delete the neighborhood
+    // If confirmed, delete neighborhood
```
We will have to issue another check on the frontend to see that the logged user is admin, right?

radu-constantin commented on apps/backend/src/tests/neighborhood_api.test.ts:
```diff
@@ -159,11 +164,11 @@ describe('Tests for creating a single neighborhood: POST /neighborhoods/:id ', (
   test('when user not logged in, unable to create neighborhood', async () => {
     const postResponse: Response = await api.post('/api/neighborhoods');
-    // console.log(postResponse?.body);
+    console.log(postResponse?.body);
```
Do we still need this? Same for line 171

nina-dk replied:
Not sure. It's been a while since I got this error, but because it's untraceable when it happens, I've left these to help with debugging if it occurs again.

radu-constantin reviewed (approved):
Excellent work, as usual. I've added a few comments and I want to mention that we need to be careful that from now on we only show requests and responses that do not have the 'INACTIVE' status.

nina-dk commented on apps/backend/src/controllers/neighborhoods.ts:
```diff
@@ -91,6 +91,8 @@ neighborhoodsRouter.post('/', middleware.userIdExtractorAndLoginValidator, catch
   res.status(201).json(newNeighborhoodWithRelatedFields);
 }));
 
+// Should probably be `put` instead
+// Join a neighborhood
```
Looks like we should be using `patch` way more often...

nina-dk commented:
> Excellent work, as usual. I've added a few comments and I want to mention that we need to be careful that from now on we only show requests and responses that do not have the 'INACTIVE' status.

I think it'd be easier and more helpful for the users to still display inactive requests/responses and simply have a different class to style them, as we will for closed requests. This way they can still view their old responses to an inactive, request, for example, but are aware that the user deactivated it or left the neighbourhood.

nina-dk commented:
@radu-constantin Turns out the bug came from adding the react-router as a dependency on the front-end. I removed it and the Outlet renders normally. I fixed all lints except two which may require more work:
- Currently, we're importing some types from `@prisma/client` and functions from `react-router` without having those packages listed as dependencies on the front-end `package.json` file. Adding them causes issues, especially with prisma. I tried it and it no longer recognized `User`, for example, as an exported member. I think the ideal solution would be to export the required types from the backend and import them from there, or figure out a way to have a shared `types.ts`. For now, I've left a warning from ESlint.
- A minor thing about unnecessary else return, which doesn't make much sense since we're actually using `else if` with a different required condition. I've made this into a warning, as well, as it does apply in some cases.

One more thing relative to the first issue, there's some overlap and bad organization of our types. I think some refactoring is due at some point.
